File: eva-core/src/main/java/io/nerv/core/enums/StatusEnumm.java
package io.nerv.core.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author: S.PKAQ
 */
@Getter
@AllArgsConstructor
public enum StatusEnumm implements BizCode{
    /**
     * 可用
     */
    ENABLE("正常", "0001"),
    UNABLE("不可用", "0000");

    /**
     * 名称
     */
    private String name;
    /**
     * 索引
     */
    private String index;
}

File: eva-core/src/main/kotlin/io/nerv/core/enums/StatusEnumm.kt
package io.nerv.core.enums

import lombok.AllArgsConstructor
import lombok.Getter

/**
 * @author: S.PKAQ
 */
enum class StatusEnumm(var value: String, var code: String)  : BizCode {
    /**
     * 可用
     */
    ENABLE("正常", "0001"),
    UNABLE("不可用", "0000");

    override fun getName(): String? {
        return this.value
    }

    override fun getIndex(): String? {
        return this.code
    }
}